app/components/star-rating/app-star-rating.component.ts

```diff
@@ -0,0 +1,56 @@
+import angular from "angular";
+
+interface StarRatingBindings {
+  numberOfStars?: number;
+  iconSize?: string;
+  onChooseRating: ({ rating: number }) => null;
+}
+
+interface IStarRating extends StarRatingBindings {
+  onClickRating: (index: number) => void;
+  onHoverRating: (index: number) => void;
```

Comment:
onClick?

Reply:
I wanted to be more verbose - we are clicking a star rating that has an index, not just clicking the root element.